dart/generate-website-screenshot/lib/main.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+Future<void> start(final req, final res) async {
+  Map<String, dynamic> data = jsonDecode(req.payload);
+
+  // Checking if url Parameter is present in the payload
+  if (!data.containsKey("url")) {
+    throw Exception("Missing parameter url.");
+  }
+
+  final url = data["url"];
+
+  // Checking if SCREENLY_API_KEY is provided in Enviornment.
+  if (req.variables['SCREENLY_API_KEY'] == null) {
+    res.json({
+      "success": false,
+      "message": "Missing Screenly API Key in environment variables",
+    });
+    return;
+  }
+
+  final api_key = req.variables['SCREENLY_API_KEY'];
+
+  var parsedUrl = Uri.parse("http://screeenly.com/api/v1/fullsize?key=${api_key}&url=${url}");
+
+  // Sending Post request to Screenly to get the screenshot.
+  var response = await http.post(parsedUrl);
+
+  // Response if the request was a success
+  if (response.statusCode == 200) {
+    Map<String, dynamic> parsedBody = jsonDecode(response.body);
+    res.json({"success": true, "screenshot": parsedBody["base64_raw"]});
+  } 
+  
+  // Response otherwise
+  else {
+    Map<String, dynamic> parsedBody = jsonDecode(response.body);
+    res.json({"success": false, "message": "Some error occured"});
```

Comment:
Let's be a little more descriptive:

```suggestion
    Map<String, dynamic> parsedBody = jsonDecode(response.body);
    final title = parsedBody['title'];
    final message = parsedBody['message'];
    res.json({"success": false, "message": '$title: $message'});
```